src/DittoMachine.sol

```diff
@@ -159,7 +162,11 @@ contract DittoMachine is ERC1155D, ERC721TokenReceiver, ERC1155TokenReceiver, Cl
                     _ERC20Contract,
                     true
                 )));
-                floorId = uint(keccak256(abi.encodePacked(floorId, index)));
+                assembly ("memory-safe") {
+                    mstore(0, floorId)
```

Comment:
it would be nice to have comments in the assembly sections where we are reusing variables / assigning an index to an id variable name just for clarification.

Reply:
done